src/languages/ruby.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  argumentMatcher,
+  conditionMatcher,
+  createPatternMatchers,
+  trailingMatcher,
+} from "../util/nodeMatchers";
+import { NodeMatcherAlternative, ScopeType, SelectionWithEditor } from "../typings/Types";
+import { SyntaxNode } from "web-tree-sitter";
+import { getNodeRange } from "../util/nodeSelectors";
+
+// Generated by the following command:
+// > curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-ruby/master/src/node-types.json \  
+// | jq '[.[] | select(.type == "_statement" or .type == "_simple_statement") | .subtypes[].type]'
+const STATEMENT_TYPES = [
+  "_expression",
+  "alias",
+  "begin_block",
+  "end_block",
+  "if_modifier",
+  "rescue_modifier",
+  "undef",
+  "unless_modifier",
+  "until_modifier",
+  "while_modifier",
+];
+
+const mapTypes = ["hash"];
+const listTypes = ["array", "string_array", "symbol_array"];
+
+const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
+    map: mapTypes,
+    list: listTypes, 
+    statement: STATEMENT_TYPES, 
+    string: "string",
+    ifStatement: "if",
+    functionCall: "call",
+    comment: "comment",
+    namedFunction: "method",
+    functionName: "method[name]",
+    anonymousFunction: "lambda",
+    regularExpression: "regex",
+    condition: conditionMatcher("*[condition]"), 
+    argumentOrParameter: argumentMatcher("lambda_parameters", "method_parameters", "block_parameters"),
+    class: "class",
+    className: "class[name]",
+    collectionKey: trailingMatcher(
+    [
+      "pair[key]",
+    ],
+    [":"]
+    ),
+    name: ["assignment[left]", "operator_assignment[left]"],
+    value: trailingMatcher(
+    [
+      "pair[value]",
+      "assignment[right]",
+      "operator_assignment[right]"
+    ],
+    [":"]
+    ),
```

Comment:
Not at a moment if you don't want to make your own finder. I want to update the matchers to use operator instead of specific text but that is still in the backlog.